docs/evaluate/temporal-cloud/pricing.mdx

```diff
@@ -29,182 +29,408 @@ tags:
   That's because Temporal Cloud counts all [RecordMarkers](https://docs.temporal.io/references/commands#recordmarker) from each Workflow Task as one action, and not _N_ actions.
   Please note:
   - Each additional Workflow Task heartbeat after counts as an additional Action.
   - Local Activities retried following a Workflow Task heartbeat count as one Action (capped at 100 Actions).
-- **Activity Heartbeat recorded.**
+- **Activity Heartbeat recorded**.
   A Heartbeat call from Activity code counts as an Action only if it reaches the [Temporal Server](/clusters#temporal-server).
   Temporal SDKs throttle [Activity Heartbeats](/encyclopedia/detecting-activity-failures#activity-heartbeat).
   The default throttle is 80% of the [Heartbeat Timeout](/encyclopedia/detecting-activity-failures#heartbeat-timeout).
   Heartbeats don't apply to Local Activities.
 
-**Schedules**
+**SCHEDULES**
 
 [Schedules](/workflows#schedule) allows you to "schedule" a Workflow to start at a particular time.
-Each execution of a Schedule will accrue three actions:
+Each execution of a Schedule accrues three actions:
 
 - **Schedule Start**.
   This accounts for two actions.
 - **Workflow started**.
   This is a single action to start the target workflow.
 
-**Export**
+**EXPORT**
 
 [Workflow History Export](/cloud/export) enables you to export closed Workflow Histories to a cloud storage sink of your choice.
 
 - **Workflow exported**.
   Each Workflow exported accrues a single action.
 
-[Reach out to our team](https://pages.temporal.io/contact-us) to get more information or to help size your number of Actions.
+</details>
+
+[Reach out](https://pages.temporal.io/contact-us) to our team for more information or to help size your number of Actions.
 
-## Storage pricing {#storage}
+### Storage {#storage}
 
-**What are the Temporal Cloud storage prices?**
+**How Workflow Storage works**
 
 A particular Workflow's execution might exist for a few seconds, a day, month, or even forever.
-Temporal collects the Workflow Execution's Event History during this time and dispatches work when necessary.
-Under this context, a Workflow Execution has only two states, open (active) or closed.
+The Temporal Service stores the Workflow Execution's [Event History](/workflows#event-history).
+Under this framework, a Workflow Execution has only two states, open (Active Storage) or closed (Retained Storage).
+
+- _Active Storage_ measures the amount of storage used by active Workflows.
+
+- When the execution of a Workflow finishes, Temporal Cloud stores Event History for a defined Retention Period, which is set by the user per Namespace.
+  This is _Retained Storage_.
+  Typical uses of Retained Storage include compliance, debugging, workload refresh, and business analytics.
+  When retention of closed Workflow histories are needed for longer than the maximum 90-day period on Temporal Cloud, we recommend using our [**Export**](/cloud/export) feature.
```

Comment:
"When retention of closed Workflow histories are needed" sounds awkward to me. How about: "When closed Workflow histories need to be retained for more than the 90-day maximum period on Temporal Cloud, we recommend using our [**Export**](/cloud/export) feature."

Reply:
Adopting as suggested.

Reply:
Also capped Histories